### PR TITLE
Move std and var tests to OpInfos

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50903 Add correction parameter to std/var
* #50902 Make at::numeric_limits non CUDA-specific
* **#50901 Move std and var tests to OpInfos**

Differential Revision: [D26083289](https://our.internmc.facebook.com/intern/diff/D26083289)

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -246,7 +246,10 @@ def test_variant_consistency_eager(self, device, dtype, op):
     # TODO WARNING: inplace x {traced, scripted} not currently tested
     @ops(op_db)
     def test_variant_consistency_jit(self, device, dtype, op):
-        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        test_backward = (
+            (dtype.is_complex and op.test_complex_grad) or
+            (dtype.is_floating_point and (not op.skip_bfloat16_grad or dtype != torch.bfloat16)))
+        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
```
Comment (peterbell10) [inline test/test_ops.py:252]: `std` and `var` aren't whitelisted for complex autograd, so `test_complex_grad` needs to never call `backward()` at all. Why wasn't this here already? Well it looks like `sigmoid` is the only other user of `test_complex_grad` and it explicitly skips these two jit tests:

https://github.com/pytorch/pytorch/blob/7fdc6a27b81d1ece894005db3b03860afd935a12/torch/testing/_internal/common_methods_invocations.py#L1608-L1612

File: test/test_ops.py [hunk 2]
```diff
@@ -249,7 +249,10 @@ def test_variant_consistency_eager(self, device, dtype, op):
     # TODO WARNING: inplace x {traced, scripted} not currently tested
     @ops(op_db)
     def test_variant_consistency_jit(self, device, dtype, op):
-        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        test_backward = (
+            (dtype.is_complex and op.test_complex_grad) or
+            (dtype.is_floating_point and (not op.skip_bfloat16_grad or dtype != torch.bfloat16)))
```
Comment (mruberry) [inline test/test_ops.py:254]: cc @Lilyjjo

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -1401,6 +1414,18 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
...
                                 device_type='cuda', dtypes=[torch.float16]),
                    )),
+    OpInfo('std',
+           dtypes=floating_types_and(),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_std_var,
+           supports_tensor_out=False,
+           test_complex_grad=False,
+           test_inplace_grad=False,
+           skips=(SkipInfo('TestGradients', 'test_fn_gradgrad'),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1424]: Would you elaborate on these two skips?

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -818,6 +818,19 @@ def sample_inputs_linalg_solve(op_info, device, dtype, requires_grad=False):
     return out
 
 
+def sample_inputs_std_var(op_info, device, dtype, requires_grad):
+    tensor_nd = torch.rand(S, S, S, device=device, dtype=dtype, requires_grad=requires_grad)
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:822]: Why rand and not make_tensor?

**Review by mruberry — COMMENTED** [review]
Hey @peterbell10!

The cleanup in test_ops.py looks great. I have a suggestion and a couple questions about the OpInfos, though. Should be simple to resolve.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (peterbell10) [reply torch/testing/_internal/common_methods_invocations.py]: I removed the first skip, after reviewing it I realised the tests passed as-is. (probably fixed by the `test_complex_grad` changes here). As for the second skip, the CPU version of `std` and `var` have only partial support for half and complex, because they sometimes call into the old TH version for all-reductions:

https://github.com/pytorch/pytorch/blob/ac0a3cc5fdaf2b559d4b96f41d7accdc5fe9c320/aten/src/ATen/native/ReduceOps.cpp#L1151-L1156

So for the `OpInfo`: `Half`, `ComplexFloat` and `ComplexDouble` are declared as unsupported on cpu but not all sample inputs will raise.

File: torch/testing/_internal/common_methods_invocations.py (hunk 4 above)
Comment (peterbell10) [reply torch/testing/_internal/common_methods_invocations.py]: Changed to `make_tensor` now.

File: test/test_ops.py (hunk 1 above)
Comment (Lilyjjo) [reply test/test_ops.py:254]: These changes to the JIT testing look good to me, it was a bad oversight. Thanks for catching it

**Comment by peterbell10** [comment]
@mruberry PTAL.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py]: That makes sense. Would you file an issue and add a comment citing it?

**Review by mruberry — APPROVED** [review]
Cool!

Would you just add a comment with an issue for the skips and also remove the std and var entries here:

https://github.com/pytorch/pytorch/blob/6dda0363bb47fbfbceb242098f63b67560540584/test/test_torch.py#L6726

Then ping me and I'll merge this.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (peterbell10) [reply torch/testing/_internal/common_methods_invocations.py]: Opened #51127

**Comment by peterbell10** [comment]
@mruberry
